lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -41,6 +41,38 @@ def object
           end
         end
 
+        # Parse a Hash or ActionController::Parameters representing a JSON API document
+        # into an ActiveRecord-ready hash.
+        # NOTE(beauby): Currently this does not handle relationships with modified keys.
+        #
+        # @param [Hash|ActionController::Parameters] document
+        # @return [Hash] ActiveRecord-ready hash
+        #
+        def self.parse(document)
+          hash = {}
+
+          hash[:id] = document['data']['id'] if document['data']['id']
```

Comment:
Rails' `from_json` and this method do different stuff. Here we are not building an instance of a model, we are just translating a json payload into a hash that can be passed to an AR model's constructor.